superset/assets/src/visualizations/nvd3/NVD3Vis.js

```diff
@@ -260,6 +264,37 @@ function nvd3Vis(element, props) {
     return types.indexOf(vizType) >= 0;
   }
 
+  function findYAxisField(metrics, publihedColumns) {
+    let columnName = '';
+
+    if (metrics && publihedColumns) {
+
+      publihedColumns.forEach((column) => {
+        metrics.forEach((d) => {
+          if (d.column.column_name ===  column) {
+            columnName = column;
+            return;
+          }
+        });
+      });
+    }
+
+    return columnName;
+  }
+
+  function findXAxisField(xField, publihedColumns) {
+    let columnName = '';
+
+    publihedColumns.forEach((column) => {
```

Comment:
(question) I don't know what are the supported browsers, but couldn't we use `.find` ?

Reply:
We support chrome at the moment and in short run all major browser in this order Firefox, safari, Edge.
All browser version would be latest and should support grid-layout css spec.
I think published Column is Array of string not dom element.

Reply:
if `publishedColumns` is Array<any>, then it should support `.find` . https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

Reply:
updated